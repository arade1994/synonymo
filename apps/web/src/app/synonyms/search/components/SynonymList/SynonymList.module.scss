.synonyms {
  margin-top: 2rem;
  padding: 1.5rem;
  border: 1px solid #dcdcdc;
  border-radius: 8px;
  width: 100%;
  max-width: 500px;
  background: #fff;
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.05);

  h2 {
    font-size: 1.5rem;
    color: #333;
    margin-bottom: 1rem;
    border-bottom: 2px solid #6c63ff;
    padding-bottom: 0.5rem;
  }

  ul {
    list-style-type: none;
    padding: 0;
  }

  li {
    background-color: #f9f9f9;
    padding: 0.75rem 1rem;
    border-radius: 6px;
    margin-bottom: 0.5rem;
    font-size: 1rem;
    color: #555;
    transition: background-color 0.2s, color 0.2s;

    &:hover {
      background-color: #6c63ff;
      color: #fff;
    }
  }
}

.noSynonyms {
  margin-top: 2rem;
  padding: 1.5rem;
  border: 1px solid #dcdcdc;
  border-radius: 8px;
  width: 100%;
  max-width: 500px;
  background: #fff;
  box-shadow: 0 4px 12px rgba(0, 0, 0, 0.05);
  text-align: center;
  color: #555;
  font-size: 1rem;
}
