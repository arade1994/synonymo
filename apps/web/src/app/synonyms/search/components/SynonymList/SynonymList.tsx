import styles from "./SynonymList.module.scss";

interface Props {
  items: string[];
  word: string;
}

const SynonymList: React.FC<React.PropsWithChildren<Props>> = ({
  items,
  word,
}) => {
  if (items.length === 0) {
    return (
      <div className={styles.noSynonyms}>
        There are no synonyms for <strong>{word}</strong>
      </div>
    );
  }

  return (
    <div className={styles.synonyms}>
      <h2>
        Synonyms for <strong>{word}</strong>
      </h2>
      <ul>
        {items.map((s) => (
          <li key={s}>{s}</li>
        ))}
      </ul>
    </div>
  );
};

export default SynonymList;
