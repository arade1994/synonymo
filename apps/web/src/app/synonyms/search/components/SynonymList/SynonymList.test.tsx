import { describe, expect, test } from "vitest";

import { render } from "@testing-library/react";

import SynonymList from "./SynonymList";

describe("SynonymList", () => {
  test("It should match a snapshot when rendered by default with no items", () => {
    const { baseElement } = render(<SynonymList items={[]} word="test" />);
    expect(baseElement).toMatchSnapshot();
  });

  test("It should match a snapshot when rendered by default with items", () => {
    const { baseElement } = render(
      <SynonymList items={["test2"]} word="test" />
    );
    expect(baseElement).toMatchSnapshot();
  });
});
