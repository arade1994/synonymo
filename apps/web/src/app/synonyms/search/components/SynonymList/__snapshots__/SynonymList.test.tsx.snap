// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SynonymList > It should match a snapshot when rendered by default with items 1`] = `
<body>
  <div>
    <div
      class="_synonyms_c0fc32"
    >
      <h2>
        Synonyms for 
        <strong>
          test
        </strong>
      </h2>
      <ul>
        <li>
          test2
        </li>
      </ul>
    </div>
  </div>
</body>
`;

exports[`SynonymList > It should match a snapshot when rendered by default with no items 1`] = `
<body>
  <div>
    <div
      class="_noSynonyms_c0fc32"
    >
      There are no synonyms for 
      <strong>
        test
      </strong>
    </div>
  </div>
</body>
`;
